ghost/core/core/server/models/post.js

```diff
@@ -1000,6 +1000,12 @@ Post = ghostBookshelf.Model.extend({
             });
         }
 
+        // If visibility is set to specific tiers, make sure we are only saving paid tiers
+        if (this.get('visibility') === 'tiers' && this.get('tiers')) {
+            const paidTiers = this.get('tiers').filter(t => t.type === 'paid');
+            this.set('tiers', paidTiers);
+        }
```

Comment:
I don't think this will work, hence why I wanted the E2E tests added. If you only specify the ids, this won't have a type property, so it won't be able to filter it. Also the frontend can pass whatever they want here - so this isn't really validating